rp2040-hal/Cargo.toml

```diff
@@ -54,6 +55,7 @@ hd44780-driver = "0.4.0"
 pio-proc = "0.2.0"
 dht-sensor = "0.2.1"
 rand = { version = "0.8.5", default-features = false }
+nostd_async = { version = "0.6.1", features = ["cortex_m"] }
```

Comment:
AFAIU Embassy opted to have its own embedded-hal impl from scratch in embassy-rp.
Is there usecases where it'd make sense for an embassy user to rely on rp2040-hal's drivers instead?

I figured demonstrating an alternative light-weight executor would make a good example.

Reply:
Yes, embassy-rp is independent from rp2040-hal. But embassy-executor should be compatible with rp2040-hal. (Also, AFAIU. It's not like I did much async embedded stuff yet.)

For learning about how async works, I very much like the idea of a light-weight stand-alone executor. However I'm not so sure if it's the best example if someone just wants to start writing an async firmware. I _think_ embassy is by far the most popular embedded async executor, and it would be best to show novices the most common way of using async. It's much easier to find help in case something goes wrong.

Anyway, that's a broader discussion and we don't need to decide about it in this pull request. For now, the examples are perfectly find.